azurelinuxagent/common/protocol/wire.py

```diff
@@ -867,29 +790,29 @@ def get_goal_state(self):
         return self._goal_state
 
     def get_hosting_env(self):
-        if self._hosting_env is None:
+        if self._goal_state is None:
             raise ProtocolError("Trying to fetch Hosting Environment before initialization!")
-        return self._hosting_env
+        return self._goal_state.hosting_env
 
     def get_shared_conf(self):
-        if self._shared_conf is None:
+        if self._goal_state is None:
             raise ProtocolError("Trying to fetch Shared Conf before initialization!")
-        return self._shared_conf
+        return self._goal_state.shared_conf
 
     def get_certs(self):
         # This property can be None in the GoalState, so not returning a ProtocolError here
-        return self._certs
+        return self._goal_state.certs
```

Comment:
This will throw an error if `self._goal_state` is None, we should handle that case

Reply:
Thanks, added